src/lib/link.spec.ts

```diff
@@ -20,6 +20,47 @@ test.serial('establish new client-connection', async (t) => {
   // TODO: ensure it is updated
 });
 
+test.serial('reuse existing connections', async (t) => {
+  const [src, dest] = await setup();
+
+  const oldLink = await Link.createWithNewConnections(src, dest);
+  const connA = oldLink.endA.connectionID;
+  const connB = oldLink.endB.connectionID;
+
+  const newLink = await Link.createWithExistingConnections(
+    src,
+    dest,
+    connA,
+    connB
+  );
+
+  // ensure the data makes sense (TODO: more?)
```

Comment:
I would propose creating a channel on oldLink first. When you reuse connection, ensure that channel is there.